addon/components/detail.js

```diff
@@ -138,11 +152,27 @@ export default Component.extend(PropTypeMixin, {
     return _.cloneDeep(bunsenView)
   },
 
+  /**
+   * Precompute all model dependencies
+   * @param {Object[]} cells - the view cells
+   * @param {Object} cellDefinitions - the view cell definitions
+   * @returns {Object} cellConfigs with  precomputed model dependencies
+   */
   @readOnly
-  @computed('renderView')
-  cellTabs () {
-    const cells = this.get('renderView.cells')
+  @computed('renderView.cells', 'renderView.cellDefinitions')
+  precomputedCellConfig (cells, cellDefinitions) {
```

Comment:
You are correct. There was nothing non-performant with the `mergedConfig` logic you had before, I just needed to precompute it so that I know all the dependencies model before it had a chance to render things.